src/org/zaproxy/zap/extension/wappalyzer/AppPattern.java

```diff
@@ -1,18 +1,30 @@
 package org.zaproxy.zap.extension.wappalyzer;
 
-import java.util.regex.Pattern;
+import com.google.re2j.Pattern;
 
 public class AppPattern {
 
-	private Pattern pattern = null;
+	private Pattern re2jPattern = null;
+	private java.util.regex.Pattern javaPattern = null;
 	private String version = null;
 	private int confidence = 100;
 	
-	public Pattern getPattern() {
-		return pattern;
+	public void setPattern(String pattern) {
+		this.javaPattern = java.util.regex.Pattern.compile(pattern, Pattern.CASE_INSENSITIVE);
+		try {
+			// This takes precedence, if it compiles
+			this.re2jPattern = Pattern.compile(pattern, Pattern.CASE_INSENSITIVE);
+		} catch (com.google.re2j.PatternSyntaxException e) {
+			// Ignore
+		}
 	}
-	public void setPattern(Pattern pattern) {
-		this.pattern = pattern;
+	/**
+	 * Returns the java version of the regex pattern - should not be used for matching in this package 
```

Comment:
It is used later. I guess for evidence? Perhaps a valid or preferred use should be mentioned here?

Reply:
Its used when searching for evidence as the core code only accepts a java Pattern - I can mention that..